#!/usr/bin/env python
PACKAGE = "sensor_fusion"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("tracker_life_time_decay", double_t, 0, "The decay to apply to object life time. this value is a linear constant to decay the life by", 10.0, 0, 1000.0)
gen.add("tracker_life_time_divider", double_t, 0, "The decay divider used to quickly remove unmatched tracks when processing", 2.0, 0, 10.0)
gen.add("tracker_life_time_threshold", double_t, 0, "The life time threshold used to purge old tracks if they have life times less than", 0.1, 0, 1.0)
gen.add("tracker_out_of_range_dist", double_t, 0, "The distance an object must be away to be considered unassociated (new)", 5.0, 0, 50.0)
gen.add("tracker_unassociated_group_dist", double_t, 0, "The distance from an unassociated object (new object) that is used to group other new objects", 2.0, 0, 20.0)
gen.add("tracker_score_threshold", double_t, 0, "The score (distance) to consider objects to be paired", 0.35, 0, 1.0)

exit(gen.generate(PACKAGE, "sensor_fusion", "SensorFusion"))